### ISSUE TITLE
[chip-test] chip_sw_clkmgr_external_clk_src_for_sw

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#16836]
### Test point name

[chip_sw_clkmgr_external_clk_src_for_sw](https://github.com/lowRISC/opentitan/blob/316831f8c2199f2288dd57ac4aeaec3334b5dfb8/hw/top_earlgrey/data/chip_testplan.hjson#L945)

### Host side component

SystemVerilog

### OpenTitanTool infrastructure implemented

None

### Contact person

@msf

### Checklist

This test is already developed. This issue tracks the enhancement for V3 as referenced in the testplan link above:
```
            Disable external clock source and verify the AST reliably falls back to the internal
            clock. Ensure the chip operates normally.
```

**Comment by andreaskurth** [comment]
Triaged for `clkmgr`. Assigning to M2.5 with https://github.com/lowRISC/opentitan/labels/Priority%3AP2 because I think we should test this functionality in a TLT.

**Comment by msfschaffner** [comment]
A simple way of carrying out this test addition would be to turn off the testbench clock right after switching back to internal clock. 
The SW test should freeze and time out if the AST does not correctly switch back to internal oscillators.

**Comment by matutem** [comment]
An even simpler way to test this is to perform cycle measurements after disabling the external clock: if the external clock was still driving then the main clock frequency would be either 48 or 96 MHz instead of 100 MHz.

**Comment by msfschaffner** [comment]
Yes that would work as well, and may be a better way of testing this than relying on a test timeout as suggested above.
